server/container/guice/jpa-common-guice/src/main/java/org/apache/james/modules/data/JPAEntityManagerModule.java

```diff
@@ -75,7 +74,7 @@ JPAConfiguration provideConfiguration(PropertiesProvider propertiesProvider) thr
                 .driverName(dataSource.getString("database.driverClassName"))
                 .driverURL(dataSource.getString("database.url"))
                 .testOnBorrow(dataSource.getBoolean("datasource.testOnBorrow", false))
-                .validationQueryTimeoutSec(dataSource.getInt("datasource.validationQueryTimeoutSec", JPAConstants.VALIDATION_NO_TIMEOUT))
+                .validationQueryTimeoutSec(dataSource.getInteger("datasource.validationQueryTimeoutSec", null))
```

Comment:
default value `JPAConstants.VALIDATION_NO_TIMEOUT` has value `-1` which is not allowed at the moment, turns it into an empty Optional.